src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,200 @@
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars "
+                + "SET is_deleted = TRUE "
+                + "WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't delete the car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT id, manufacturer_id, model "
+                + "FROM cars "
+                + "JOIN cars_drivers cd "
+                + "ON cars.id = cd.car_id "
+                + "WHERE cd.driver_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+
+            List<Car> carsList = new ArrayList<>();
+            while (resultSet.next()) {
+                Car car = getCar(resultSet);
+                carsList.add(car);
+            }
+            return carsList;
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't get all driver's cars by their id " + driverId, e);
+        }
+    }
+
+    private void addCarDriversRelationship(Car car) {
+        String query = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getId());
+            List<Driver> driversList = car.getDriversList();
+            for (Driver driver : driversList) {
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't insert new cars_drivers relationship by " + car, e);
+        }
+    }
+
+    private void deleteCarDriversRelationship(Long carId) {
+        String query = "DELETE FROM cars_drivers WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, carId);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't delete cars_drivers relationship by id " + carId, e);
+        }
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        ManufacturerDao manufacturerDao = new ManufacturerDaoImpl();
```

Comment:
you shouldn't use any other dao implementations in this class, you should get a manufacturer info using queries created in this class